ldap.go

```diff
@@ -18,146 +17,39 @@ const (
-}
-
-func (x *LdapAuthenticator) Close() {
-	if x.con != nil {
-		x.con.Close()
-		x.con = nil
-	}
-}
-
-func (x *ldapUserStore) SetPassword(userId UserId, password string) error {
-	return ErrUnsupported
-}
-
-func (x *ldapUserStore) ResetPasswordStart(userId UserId, expires time.Time) (string, error) {
-	return "", ErrUnsupported
-}
-
-func (x *ldapUserStore) ResetPasswordFinish(userId UserId, token string, password string) error {
-	return ErrUnsupported
-}
-
-func (x *ldapUserStore) CreateIdentity(email, username, firstname, lastname, mobilenumber, password string) (UserId, error) {
-	// Insert into user store
-	if tx, etx := x.db.Begin(); etx == nil {
-		// Check if the user exists (is not archived)
-		row := tx.QueryRow("SELECT email FROM authuserstore WHERE LOWER(email) = $1 AND archived = false", CanonicalizeIdentity(email))
-		scanErr := row.Scan()
-		if strings.Index(scanErr.Error(), "no rows in result set") == -1 {
-			tx.Rollback()
-			scanErr = ErrIdentityExists
-			return NullUserId, scanErr
-		}
-
-		if _, eCreateUserStore := tx.Exec(`INSERT INTO authuserstore (email, username, firstname, lastname, mobile, archived) VALUES ($1, $2, $3, $4, $5, $6)`, email, username, firstname, lastname, mobilenumber, false); eCreateUserStore != nil {
-			tx.Rollback()
-			return NullUserId, eCreateUserStore
-		}
-
-		// Get user id
-		var userId int64
-		row = tx.QueryRow(`SELECT userid FROM authuserstore WHERE email = $1`, email)
-		if scanErr := row.Scan(&userId); scanErr != nil {
-			tx.Rollback()
-			return NullUserId, scanErr
-		}
-
-		if eCommit := tx.Commit(); eCommit != nil {
-			return NullUserId, eCommit
-		}
-		return UserId(userId), nil
-	} else {
-		return NullUserId, etx
-	}
-}
-
-func (x *ldapUserStore) RenameIdentity(oldIdent, newIdent string) error {
-	return ErrUnsupported
-}
 
-func (x *ldapUserStore) GetIdentities() ([]AuthUser, error) {
-	rows, err := x.db.Query(`SELECT userid, email, username, firstname, lastname, mobile FROM authuserstore WHERE (archived = false OR archived IS NULL) ORDER BY username`)
+	con, err := NewLDAPConnect(x.config)
+	defer con.Close()
 	if err != nil {
-		return []AuthUser{}, err
-	}
-	defer rows.Close()
-	result := make([]AuthUser, 0)
-	type sqlUser struct {
-		userid       sql.NullInt64
-		email        sql.NullString
-		username     sql.NullString
-		firstname    sql.NullString
-		lastname     sql.NullString
-		mobilenumber sql.NullString
-	}
-	for rows.Next() {
-		user := sqlUser{}
-		if err := rows.Scan(&user.userid, &user.email, &user.username, &user.firstname, &user.lastname, &user.mobilenumber); err != nil {
-			return []AuthUser{}, err
-		}
-		result = append(result, AuthUser{UserId(user.userid.Int64), user.email.String, user.username.String, user.firstname.String, user.lastname.String, user.mobilenumber.String})
+		return err
 	}
-	if rows.Err() != nil {
-		return []AuthUser{}, rows.Err()
+	err = con.Bind(fmt.Sprintf(`%v@%v`, identity, x.config.LdapDomain), password)
+	if err != nil {
```

Comment:
If we're sending a specific error to be displayed on the front end, is it not viewed as a security risk? E.g.: Allowing any potential malicious user to know that a password failed on an existing user gives him sufficient knowledge to try a brute force login for that user. Yes, we can have other measures to prevent brute-forcing, but isn't this the first line of defence? A single, blanketed error on the front-end is kind of industry-standard security these days.